trace.go

```diff
@@ -50,18 +50,24 @@ func IsDebug() bool {
 // Wrap takes the original error and wraps it into the Trace struct
 // memorizing the context of the error.
 func Wrap(err error, args ...interface{}) Error {
+	return InternalWrap(1, err, args...)
+}
+
+// InternalWrap wraps the original error into the Trace struct with the ability
```

Comment:
I don't think this is necessary - the trail can simply use `trace.XXX` to wrap the errors with corresponding trace error types.